app/src/main/cpp/src/components/transport_manager/src/bluetooth_le/ble_control_protocol.cc

```diff
@@ -0,0 +1,107 @@
+#include "transport_manager/bluetooth_le/ble_control_protocol.h"
+
+#include "utils/logger.h"
+#include "utils/jsoncpp_reader_wrapper.h"
+#include "utils/helpers.h"
+
+#include <map>
+#include <algorithm>
+
+namespace transport_manager {
+namespace transport_adapter {
+
+SDL_CREATE_LOG_VARIABLE("TransportManager")
+
+    namespace {
+        std::map<BleProtolActions, std::string> mapActionNames = {
+        {BleProtolActions::ON_DEVICE_CONNECTED, "ON_DEVICE_CONNECTED"},
+        {BleProtolActions::ON_DEVICE_DISCONNECTED, "ON_DEVICE_DISCONNECTED"}};
+
+        const char cDefaultValue[] = "_";
+        const char cActionKey[] = "action";
+        const char cNameKey[] = "name";
+        const char cAddressKey[] = "address";
+        const char cParamsKey[] = "params";
+    }
+
+    BleProtolActions BleControlProtocol::GetMessageActionType(const MsgData& message) {
+
+        utils::JsonReader reader_;
+        Json::Value value;
+        
+        if(reader_.parse((char*)(message.data()), &value)) {
+
+            auto action = value.get(cActionKey, cDefaultValue);
+
+            if(action.asString() != cDefaultValue) {
+                return GetActionFromString(action.asString());
+            }
+
+        }
+
+        return BleProtolActions::INVALID_ENUM;
+    }
+
+    BleProtolActions BleControlProtocol::GetActionFromString(const std::string& str) {
+        
+        BleProtolActions result = BleProtolActions::INVALID_ENUM;
+
+        auto findResult = std::find_if(mapActionNames.begin(), mapActionNames.end(),
+         [&](const std::pair<BleProtolActions, std::string> &pair) {
```

Comment:
```suggestion
         [&str](const std::pair<BleProtolActions, std::string> &pair) {
```

Reply:
fixed in 22caff304f02304d55c097802b850d8bca25a370